dask/bytes/core.py

```diff
@@ -117,10 +117,15 @@ def read_bytes(
                 comp = compression
             if comp is not None:
                 raise ValueError(
-                    "Cannot do chunked reads on compressed files."
+                    "Cannot do chunked reads on compressed files. "
                     "To read, set blocksize=None"
                 )
             size = fs.info(path)["size"]
+            if size is None:
+                raise ValueError(
+                    "Backing filesystem doesn't provide file size information, "
```

Comment:
Perhaps *didn't* provide, since it might be something general to the FS, or it might be just for this particular file or state.